server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -346,10 +344,13 @@ export function configureWebSocketServices(
             connectDocument(connectionMessage).then(
                 (message) => {
                     socket.emit("connect_document_success", message.connection);
-                    socket.emitToRoom(
-                        getRoomId(roomMap.get(message.connection.clientId)),
-                        "signal",
-                        createRoomJoinMessage(message.connection.clientId, message.details));
+                    const room = roomMap.get(message.connection.clientId);
+                    if (room) {
```

Comment:
Do we need to add some `else` condition here? I'm assuming that before, if `roomMap.get(message.connection.clientId)` was `undefined`, `getRoomId` in line 50 would try to access properties of `undefined` and then we would get an error, captured in line 354 above. Now, since we check `if (room)`, I believe no error is thrown in that case anymore. So in case `room` happens to be undefined, do we need an `else` statement to handle such condition?

Reply:
Based on the code, it should not be possible for `room` to be `undefined`. I don't see anything that deletes things from `roomMap.` So we could add a non null assertion for when `room` is passed into `getRoomId` but it feels safer to use an if statement here just incase `connectDocument` does something weird, such as not setting the client id in the room map.
In any case, I don't think an else statement is needed since this logic is just for signaling. These signals are more of a best effort thing rather than something guaranteed